jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/TreeTrie.java

```diff
@@ -276,7 +276,31 @@ private V getBest(Node<V> node, byte[] b, int offset, int len)
     @Override
     public boolean isEmpty()
     {
-        return keySet().isEmpty();
+        return isEmpty(_root);
+    }
+
+    private boolean isEmpty(Node<V> t)
+    {
+        if (t != null)
+        {
+            if (t._key != null)
+                return false;
+
+            for (int i = 0; i < INDEX; i++)
```

Comment:
Is there some test coverage for this logic?

Reply:
Yup, the first loop is 100% covered by the 2 new test cases.
The second loop executes, but nothing can trigger the internal isEmpty() logic with the test cases as it stands.